hs_core/templatetags/hydroshare_tags.py

```diff
@@ -238,3 +248,67 @@ def normalize_human_name(name):
 def display_name_to_class(value):
     """ Converts an aggregation display name to a string that is usable as a CSS class name """
     return value.replace(" ", "_").lower()
+
+
+@register.filter
+def creator_json_ld_element(crs):
+    """ return json ld element for one creator for schema.org script embedded on resource landing page"""
+    crs_array = []
+    for cr in crs:
+        cr_dict = {}
+        urls = []
+        if cr.email:
+            cr_dict["email"] = cr.email
+        if cr.address:
+            cr_dict["address"] = {
+                "@type": "PostalAddress",
+                "streetAddress": cr.address
+            }
+        if cr.name:
+            cr_dict["@type"] = "Person"
+            cr_dict["name"] = name_without_commas(cr.name)
+            if cr.organization:
+                affl_dict = {
+                    "@type": "Organization",
+                    "name": cr.organization
+                }
+                cr_dict["affiliation"] = affl_dict
+        else:
+            cr_dict["@type"] = "Organization"
+            cr_dict["name"] = cr.organization
+
+        if cr.description:
+            if cr.name:
+                # append www.hydroshare.org since schema.org script is only embedded in production
+                urls.append("https://www.hydroshare.org" + cr.description)
+            else:
+                # organization
+                urls.append(cr.description)
+        if cr.homepage:
+            urls.append(cr.homepage)
+        if cr.identifiers:
+            for k in cr.identifiers:
+                urls.append(cr.identifiers[k])
+        if len(urls) == 1:
+            cr_dict['url'] = urls[0]
+        elif len(urls) > 1:
+            cr_dict['url'] = urls
+        crs_array.append(cr_dict)
+    # reformat json dumped str a bit to fix the indentation issue with the last bracket
+    default_dump = dumps(crs_array, sort_keys=True, indent=4)
+    format_dump = '{}    {}'.format(default_dump[:-1], default_dump[-1])
+    return format_dump
+
+
+@register.filter
+def bag_checksum(res):
```

Comment:
Same with this here, accessing the bag_checksum feels like it should be an easy call for checksum on the resource object instead of looking it up in extra_data.  The complexity should be encapsulated in the resource object.